### PR TITLE
Re-add and implement schedule list

### PR DESCRIPTION
**Pull request by jasperfung**
There's still an issue around prematurely showing a "No schedule uploaded" message, caused by our redundant data fetching (we apparently make 2+ calls whenever loading the logged-in user's profile). Looking into this after implementing schedule alarm creation.

### CODE REVIEW

**Comment by divad12** [comment]
Cool!

File: src/com/uwflow/flow_android/adapters/ProfileScheduleAdapter.java [hunk 1]
```diff
@@ -0,0 +1,207 @@
...
+import com.uwflow.flow_android.fragment.CourseFragment;
+import com.uwflow.flow_android.util.CalendarHelper;
+import com.uwflow.flow_android.util.CourseUtil;
+
+import java.text.SimpleDateFormat;
+import java.util.*;
+
+
+public class ProfileScheduleAdapter extends BaseExpandableListAdapter {
+    private Map<String,List<ScheduleCourse>> mScheduleMap;
```
Comment (divad12) [inline src/com/uwflow/flow_android/adapters/ProfileScheduleAdapter.java:28]: Can you comment what this is a map from and to?

File: src/com/uwflow/flow_android/adapters/ProfileScheduleAdapter.java [hunk 2]
```diff
@@ -0,0 +1,207 @@
...
+        final String location = String.format("%s %s", locationBuilding, locationRoom);
+        second.setText(String.format("%s - %s    %s",
+                timeFormat.format(startDate),
+                timeFormat.format(endDate),
+                location));
+
+        convertView.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                // TODO: Add an alarm for the current course
```
Comment (divad12) [inline src/com/uwflow/flow_android/adapters/ProfileScheduleAdapter.java:157]: Remove this TODO?

File: src/com/uwflow/flow_android/fragment/ProfileScheduleFragment.java [hunk 3]
```diff
@@ -53,21 +65,43 @@ public View onCreateView(LayoutInflater inflater, ViewGroup container,
...
+        });
 
-        mBtnShare.setEnabled(false);
         mBtnShare.setOnClickListener(this);
+        mImageSchedule.setOnClickListener(this);
 
         // call this before setting up the receiver
-        populateData(/* isFromServer */ false);
-        profileScheduleReceiver = new ProfileScheduleReceiver();
+//        populateData(/* isFromServer */ false);
```
Comment (divad12) [inline src/com/uwflow/flow_android/fragment/ProfileScheduleFragment.java]: Remove commented-out code?

File: src/com/uwflow/flow_android/network/FlowDatabaseLoader.java [hunk 4]
```diff
@@ -212,11 +212,6 @@ public Void call() throws Exception {
                                     for (ScheduleCourse sc : scheduleCourses.getScheduleCourses()) {
                                         sc.setScheduleUrl(scheduleCourses.getScreenshotUrl());
                                         userCourseSchedule.createOrUpdate(sc);
-
-                                        // FIXME(david): This is a major hack to just save the first schedule item,
-                                        //     because this loop takes a long time, causes thrashing, and we don't use
-                                        //     any of the schedule item data yet.
-                                        break;
```
Comment (divad12) [inline src/com/uwflow/flow_android/network/FlowDatabaseLoader.java:219]: Ha, sorry about that. Hope this wasn't too difficult to track down. :stuck_out_tongue:

File: src/com/uwflow/flow_android/adapters/ProfileScheduleAdapter.java (hunk 2 above)
Comment (jasperfung) [inline src/com/uwflow/flow_android/adapters/ProfileScheduleAdapter.java:157]: Removed in the subsequent commit when the Alarm is implemented.

File: src/com/uwflow/flow_android/network/FlowDatabaseLoader.java (hunk 4 above)
Comment (divad12) [inline src/com/uwflow/flow_android/network/FlowDatabaseLoader.java:219]: (I have a Trello card for investigating other ways to make first-time load for fresh installs on crappy phones work better.)

Comment (jasperfung) [inline src/com/uwflow/flow_android/network/FlowDatabaseLoader.java:219]: Lol it definitely took a while. It's all good. :)

**Comment by divad12** [comment]
LGTM!

**Pull request merged**